### ISSUE TITLE
Nixpkgs Sfepy recipe - fast test suite

### ISSUE THREAD

**Issue by wd15** [sfepy/sfepy#552]
I'm trying to finish off a [PR for Sfepy in Nixpkgs](https://github.com/NixOS/nixpkgs/pull/67788). We've found that the full test suite is quite slow O(minutes). Could @rc recommend a subset of tests that is both representative and runs in O(seconds).

Would it be possible to add `--fast` flag to run_tests.py for a more permanent solution?

**Comment by rc** [comment]
Yes. adding the `--fast` flag is a good idea. I guess most of the `test_input_*.py` tests can be omitted from the fast mode, need to think about other. Some modifications might be needed. I will try the flag next week.

**Comment by wd15** [comment]
[Here](https://github.com/NixOS/nixpkgs/blob/6271f789126a74482f727ed58d54aae2bbdae9c7/pkgs/development/python-modules/sfepy/default.nix#L37) are the list of tests removed from the Nixpkgs Sfepy recipe to make things speedier.

**Comment by rc** [comment]
Thanks for the list. I am wondering if, instead of improving the custom test runner, going with #363 would be better in the long run. Converting all the tests for use with pytest would be a lot of tedious work, but once it's done, things like marking slow tests would be easy.

**Comment by wd15** [comment]
> Thanks for the list. I am wondering if, instead of improving the custom test runner, going with #363 would be better in the long run. Converting all the tests for use with pytest would be a lot of tedious work, but once it's done, things like marking slow tests would be easy.

Absolutely makes sense if the run_tests.py stuff is all going away soon. Regarding switching over to pytest, I would recommend providing a way for users to test an installed version of sfepy. I don't like the way virtually all Python packages require a clone in order to test. I've done this with [pymks](https://github.com/materialsinnovation/pymks/blob/develop/pymks/__init__.py#L19). Also put the tests directory as part of the distribution, but with no `__init__.py`.

**Comment by rc** [comment]
As a matter of fact, `sfepy-run run_tests` allows running tests of the installed sfepy, so I would like to preserve this with pytest.
